spring-integration-core/src/main/java/org/springframework/integration/support/management/LifecycleMessageSourceManagement.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management;
+
+import org.springframework.context.Lifecycle;
+import org.springframework.integration.endpoint.MessageSourceManagement;
+
+/**
+ * An extension to {@link LifecycleMessageSourceMetrics} for sources that support
+ * max fetch size.
```

Comment:
Wouldn't it be better like: `for source that support MessageSourceManagement.` ?

Just because it is true via this:
```
 * Message sources implementing this interface have additional properties that
 * can be set or examined using JMX.
```
in the `MessageSourceManagement`